workitems/l2-transaction-fees/l2-transaction-fees-v1.0-psd01.md

```diff
@@ -196,55 +266,835 @@ Note that requirements are uniquely numbered in ascending order within each requ
+
+* Define the meaning of a transaction fee and its components
+* Define the roles relevant to transaction fees
+* Define transaction types and their relationship to transaction fees
+
+Note, that all definitions can be found in the [Glossary](#12-glossary)
+
+Subsequently, the specification section will define requirements around fee transparency, fee estimation, final fees, and how, when and where fees are communicated to the different roles. The specification will also discuss requirements around responsibilities and accountability of roles towards one another.
+
+## 2.1 Definition of Transaction Fee and its Components
+###### L2TFSPECDEF
+
+This document defines Transaction Fee as the fee in a Layer 2 (L2) network or protocol token to be paid by a Transaction Originator or Transaction Sender comprised of the sum of a Base Fee, an Execution Fee and a Priority Fee.
+
+Note that this document defines as Transaction as a digitally signed message sent from an L2 account that contains instructions and data that result in an L2 state transition. Also note that this document defines State and State Transition as follows:
+
+* State: The set of all accounts on an L2 that are mapped to one another using a cryptographic data structure.
+* State Transition: An event that deterministically changes one or more accounts in the set of all accounts that comprise the complete state of an L2.
+
+Further note that this document defines Account as a unit that is minimally comprised of a unique account identifier, a deterministic ordering parameter, also referred to as a nonce, a balance of a Layer 1, Layer 2 or Sidechain unit of exchange, also referred to as a protocol token. Changes to an account are controlled by a unique cryptographic public and private key pair. There are typically additional account elements referring to instructions and data associated to the account that determine account changes.
+
+In formula form the Transaction Fee is given as:
+
+**Transaction Fee = Base Fee + Execution Fee + Priority Fee**
+
+The different components are defined as follows:
+
+* **Base Fee:** The minimum amount of Gas or an L2 gas equivalent unit of compute and storage consumption required to include a transaction on an L2.
+* **Execution Fee:** A fee to be paid by the Transaction Originator or Transaction Sender sufficient to cover both the L2 and Layer 1 transaction fees.
+* **Priority Fee:** To be paid by the Transaction Originator or Transaction Sender to an L2 sequencer to obtain a desired slot for its transaction in a new block. Note that the exact position of a transaction, a slot, in a new block may be determined by factors such as time-stamp, or minimization of Maximal Extractable Value (MEV) of a block in addition to a Priority Fee. And that MEV is defined as the maximum value that can be extracted from block production in excess of the standard block reward and gas fees by including, excluding, and changing the order of transactions in a block.
```

Comment:
```suggestion
* **Priority Fee:** To be paid by the Transaction Originator or Transaction Sender likely to an L2 sequencer to obtain a desired slot for its transaction in a new block. Note that the exact position of a transaction, a slot, in a new block may be determined by factors such as time-stamp, or minimization of Maximal Extractable Value (MEV) of a block in addition to a Priority Fee. And that MEV is defined as the maximum value that can be extracted from block production in excess of the standard block reward and gas fees by including, excluding, and changing the order of transactions in a block.
```

(see comment above)

Reply:
@DZGoldman reworded the 1st sentence:

**Priority Fee:** Paid by the Transaction Originator or Transaction Sender to obtain a desired slot for its transaction in a new block.